datatable/java/datatable/src/main/java/io/cucumber/datatable/DataTableTypeRegistry.java

```diff
@@ -26,69 +26,79 @@
+            Type optionalByte = new TypeReference<Optional<Byte>>() {}.getType();
+            defineDataTableType(new DataTableType(optionalByte, optionalIfAbsent(byteTableCellTransformer)));
+        }
+        {
+            TableCellTransformer<Short> shortTableCellTransformer = applyIfPresent(Short::decode);
+            defineDataTableType(new DataTableType(Short.class, shortTableCellTransformer));
+            defineDataTableType(new DataTableType(short.class, shortTableCellTransformer));
+            Type optionalShort = new TypeReference<Optional<Short>>() {}.getType();
+            defineDataTableType(new DataTableType(optionalShort, optionalIfAbsent(shortTableCellTransformer)));
+        }
+        {
+            TableCellTransformer<Integer> integerTableCellTransformer = applyIfPresent(Integer::decode);
+            defineDataTableType(new DataTableType(Integer.class, integerTableCellTransformer));
+            defineDataTableType(new DataTableType(int.class, integerTableCellTransformer));
+            Type optionalInteger = new TypeReference<Optional<Integer>>() {}.getType();
+            defineDataTableType(new DataTableType(optionalInteger, optionalIfAbsent(integerTableCellTransformer)));
+        }
+        {
+            TableCellTransformer<Long> longTableCellTransformer = applyIfPresent(Long::decode);
+            defineDataTableType(new DataTableType(Long.class, longTableCellTransformer));
+            defineDataTableType(new DataTableType(long.class, longTableCellTransformer));
+            Type optionalLong = new TypeReference<Optional<Long>>() {}.getType();
+            defineDataTableType(new DataTableType(optionalLong, optionalIfAbsent(longTableCellTransformer)));
+        }
+        {
+            TableCellTransformer<Float> floatTableCellTransformer = applyIfPresent(numberParser::parseFloat);
+            defineDataTableType(new DataTableType(Float.class, floatTableCellTransformer));
+            defineDataTableType(new DataTableType(float.class, floatTableCellTransformer));
+            Type optionalFloat = new TypeReference<Optional<Float>>() {}.getType();
+            defineDataTableType(new DataTableType(optionalFloat, optionalIfAbsent(floatTableCellTransformer)));
+        }
+        {
+            TableCellTransformer<Double> doubleTableCellTransformer = applyIfPresent(numberParser::parseDouble);
+            defineDataTableType(new DataTableType(Double.class, doubleTableCellTransformer));
+            defineDataTableType(new DataTableType(double.class, doubleTableCellTransformer));
+            Type optionalDouble = new TypeReference<Optional<Double>>() {}.getType();
+            defineDataTableType(new DataTableType(optionalDouble, optionalIfAbsent(doubleTableCellTransformer)));
+        }
     }
 
     private static <R> TableCellTransformer<R> applyIfPresent(Function<String, R> f) {
         return s -> s == null ? null : f.apply(s);
```

Comment:
I'm wondering why we're not checking for the empty string here.

Reply:
I think empty string is completely fine for stringTableCellTransformer, but it is not fine for others.
Now we throw exception for this case:
```
DataTable table = parse("",
                "| header1   | header2   |",
                "| 1              |                 |"
        );
converter.toMaps(table, String.class, byte.class);

>> io.cucumber.datatable.CucumberDataTableException: 'java.util.List<java.util.List<byte>>' could not transform
      | 1 | [empty] |
```
I think we can safely change this to null if the string is empty.